node/src/components/linear_chain.rs

```diff
@@ -98,29 +100,76 @@ impl<I: Display> Display for Event<I> {
                 height,
                 block.is_some()
             ),
+            Event::GetBlockForFinalitySignaturesResult(block_hash, maybe_block) => {
+                write!(
+                    f,
+                    "linear chain get-block-for-finality-signatures-result for {} found: {}",
+                    block_hash,
+                    maybe_block.is_some()
+                )
+            }
         }
     }
 }
 
 #[derive(DataSize, Debug)]
 pub(crate) struct LinearChain<I> {
-    /// A temporary workaround.
-    // TODO: Refactor to proper LRU cache.
-    linear_chain: Vec<Block>,
+    /// The most recently added block.
+    // TODO: Refactor to proper LRU cache. Or read from storage?
```

Comment:
This TODO can be removed probably. Or at least the part about LRU since it's already a LRU cache with size=1.

Reply:
I'll remove the whole TODO, since the storage part is already duplicated with other todos.